src/Controller/IndexController.php

```diff
@@ -85,35 +85,38 @@ public function search(Request $request): Response
             return $this->redirectToRoute('index_page');
         }
 
-        $searchStr = trim($request->request->get('search_str', ''));
-        // pirma patikrinam ar ieškoma pagal įrankio kodą
-        if (is_numeric($searchStr) && (mb_strlen($searchStr) == 6)) {
-            $tool = $this->toolsRepo->findOneBy(['code' => $searchStr]);
-            if ($tool) {
-                return $this->render('tool.html.twig',
-                    [
-                        'tool'       => $tool,
-                        'search_str' => $searchStr,
-                    ]
-                );
-            }
+        $code = trim($request->request->get('search_str', ''));
+        // check if search string is more or equal to 6 and if its code
+        $endcode = substr($code, -6, 6);
+        $startcode = substr($code, 0, 6);
+        if (mb_strlen($code) >= 6 && (is_numeric($endcode) || is_numeric($startcode))) {

```

Comment:
What is the use case for this kind of check? (checking 6 chars from both start and end of search string)

Reply:
Its support for barcode scanner. Basically even if user has something in the search input, on barcode scanner action we focus input, enter barcode and press enter automatically, so with this check I'm just eliminating everything else that is not tool barcode.

Reply:
Understood. Is it possible to clean up search input field before entering the barcode on the scanner action? I think it would make a cleaner solution.
If not then current logic works too.

Reply:
Havent found fast and nice way to do it 😄 In order to clean up you have to prevent default on input function and barcode scanner was working out of the box just by listening on keypress event so we decided to make this check.

Reply:
Alright, good to go then.